### PR TITLE
feat(auth): implement auth0 IDP 

### PR DESCRIPTION
**Pull request by ProgaPanda**
## Description [[STORY LINK]](https://app.asana.com/0/750497957484456/1201473608206766/f)

This PR adds auth0 as an additional IDP, including a new exported Button from the library that can be used both in Register and Login containers.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Tested changes locally.
- [ ] Updated documentation.

## Screenshots 

https://user-images.githubusercontent.com/6924756/149520062-4c77af1d-04aa-4d5f-ac1c-c8ba7df3f2e0.mp4

### CODE REVIEW

File: src/Components/common/Auth0LoginButton/Auth0LoginButton.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        if (error) {
+          return onFailure(error);
+        }
+
+        onSuccess(authResult);
+      }
+    );
+  }
+
+  const onSuccess = (authResult) => {
```
Comment (muubar) [inline src/Components/common/Auth0LoginButton/Auth0LoginButton.js:47]: we usually have this kind of logic as part of the container, why did we include it here in this case?

**Review by Marwan-Pelcro — APPROVED** [review]
LGTM!

**Review by muubar — APPROVED** [review]
LGTM, I think we should spend a bit of time thinking if we should move the logic in the button to the login and register containers instead, so that its consistent with other buttons

File: src/Components/common/Auth0LoginButton/Auth0LoginButton.js (hunk 1 above)
Comment (ProgaPanda) [reply src/Components/common/Auth0LoginButton/Auth0LoginButton.js:47]: What I'm doing is dispatching the login success event to the container to handle the auth logic. what the button is doing in onSuccess is basically extracting the user data from the auth0 sdk.

**Pull request merged**